### DISCUSSION TITLE
Sample code not showing data (NCT6687D), but is shown by forms app

### DISCUSSION THREAD

**Discussion by EtiennePeeters** [LibreHardwareMonitor/LibreHardwareMonitor#886] [Q&A]
Hi,

The sample code is not showing sensors from my motherboard MSI MAG Z690 TORPEDO (MS-7D32), with the Nuvoton NCT6687D chip.
I have tried the code in a .NET 6.0 and .NET Framework 4.7.2 project. But that does not seem to make any difference.
While debugging I can see that the list of `SubHardware` is always empty for my mainboard.

Output: 
![sample code output](https://user-images.githubusercontent.com/7521543/201514879-4b5465f7-2ebf-431f-8fbc-1f8fa62ac44e.jpg)

I saw that the [Nuvoton NCT6687D was added](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/288) about two years ago. 
And the forms app is showing the data from the chip, so it all seems to be working correctly:
![App output](https://user-images.githubusercontent.com/7521543/201514902-8e789a61-3592-4cf8-9ba4-387ce3a842af.jpg)

I assumed all data shown by the forms app would be available through the Lib.

How can I get this data in C#? 
Thanks!

**Comment by PhyxionNL** [comment]
You need to update the hardware, check repo on how that's done. Also don't run LHM and own app together.

**Comment by EtiennePeeters** [comment]
Thanks for your quick reply! But the values in the Forms UI seem fine. Only in the sample code the subhardware from the motherboard was missing. I was wondering what the difference was between the UI and the sample code. Turns out there is no difference. 

All I needed to do was run the sample code with elevated permissions! I found out when I tried to run LibreHardwareMonitor from visual studio. Apparently elevated permissions are required to access the motherboard chip.

For anyone wondering how to make your .NET 6 console app run request admin permissions:
Add an `app.manifest` with and change the `requestedExecutionLevel` to `requireAdministrator`.
